core/src/main/java/org/mule/DefaultMessageCollection.java

```diff
@@ -231,17 +227,16 @@ public synchronized void setPayload(Object payload, DataType dataType)
     }
 
     /**
-     * Applies the {@link org.mule.api.MuleMessage#getPayload(Class)} call to every message in the collection and returns a
-     * {@link java.util.List} of results.
+     * Applies the {@link TransformationService#getPayload(MuleMessage, DataType)} call to every message in the
+     * collection and returns a {@link java.util.List} of results.
      *
      * {@inheritDoc}
      */
-    @Override
     public Object getPayload(Class outputType) throws TransformerException
     {
         if (invalidatedPayload)
         {
-            return super.getPayload(outputType);
+            return muleContext.getTransformationService().getPayload(this, outputType);
```

Comment:
I need to remove these and move this logic to TransformationService. Missed this completely so thanks! Surprised no tests were affected..